### PR TITLE
Best adventure log

### LINKED ISSUE #64: Best Adventure Log
When an admin is at a GameSession Show Page, they can see all of the Adventure Logs for that game session. 

They can designate one and only one of the logs as the best log for that session, and this adventure log will now be displayed first, with some sort of icon or tag noting that it was the best. 

The active character's show page will display how many best adventure logs that this character has received.

### PR DESCRIPTION
**Pull request by DanielEFrampton**
## Issues

Resolves #64 

## Changes

- Admin can mark an adventure log as best, which displays it at the top with a medal and tag.
- Admin can unmark a best adventure log, which puts it back in its place.
- Admin can mark a different log as best, which moves it up puts the previous best back in its place.
- Users and visitors can see the best adventure log but not mark or unmark them.
- **New**: Character show page shows number of times character has been awarded best log

## Screenshots

- Logged in as Admin, before marking a log as best:
  Screen Shot 2020-08-09 at 2 28 58 PM

- Logged in as Admin, after marking a log as best:
Screen Shot 2020-08-09 at 2 29 09 PM

- Logged in as normal user, after log marked as best:
Screen Shot 2020-08-09 at 2 29 38 PM

- Times awarded best log for both characters:
Screen Shot 2020-08-10 at 10 58 17 PM

Screen Shot 2020-08-10 at 10 58 40 PM

## State of tests

```
Finished in 9.39 seconds (files took 3.76 seconds to load)
178 examples, 0 failures

Coverage report generated for RSpec to /Users/dframpton/code/projects/cobalt_reserve/coverage. 1973 / 1973 LOC (100.0%) covered.
```

## GIF tax

luke skywalker getting medal

### CODE REVIEW

File: app/controllers/adventure_logs_controller.rb [hunk 1]
```diff
@@ -17,12 +17,25 @@ def create
     end
   end
 
+  def update
+    previous = GameSession.find(game_session_id).best_adventure_log
+    previous.update(best: false) if previous && previous.id != adventure_log_id
```
Comment (mikedao) [inline app/controllers/adventure_logs_controller.rb:22]: This is a really good approach. When I was thinking about this in my head, I was planning on just setting all adventure logs for that session and then setting them to not best and then setting the one I wanted as the best.

**Comment by mikedao** [comment]
Also the ribbon for best adventure log is GREAT.

File: app/controllers/adventure_logs_controller.rb (hunk 1 above)
Comment (DanielEFrampton) [reply app/controllers/adventure_logs_controller.rb:22]: Thanks! I started out that way.

**Comment by DanielEFrampton** [comment]
> It looks good - can you add in under the character show page, the number of best logs they have written?

Yes, would love to, I was thinking that'd be a fun addition.

**Pull request merged**